src/data/nat/prime.lean

```diff
@@ -1102,21 +1102,40 @@ end nat
 namespace nat
 
 /-- The only prime divisor of positive prime power `p^k` is `p` itself -/
-lemma prime_pow_prime_divisor {p k : ℕ} (hk : 0 < k) (hp: prime p) :
+lemma prime_pow_prime_divisor {p k : ℕ} (hk : k ≠ 0) (hp : prime p) :
   (p^k).factors.to_finset = {p} :=
-by rw [hp.factors_pow, list.to_finset_repeat_of_ne_zero hk.ne']
+by rw [hp.factors_pow, list.to_finset_repeat_of_ne_zero hk]
 
-lemma mem_factors_mul_of_pos {a b : ℕ} (ha : 0 < a) (hb : 0 < b) (p : ℕ) :
+lemma mem_factors_mul_of_ne_zero {a b : ℕ} (ha : a ≠ 0) (hb : b ≠ 0) (p : ℕ) :
   p ∈ (a * b).factors ↔ p ∈ a.factors ∨ p ∈ b.factors :=
 begin
-  rw [mem_factors (mul_pos ha hb), mem_factors ha, mem_factors hb, ←and_or_distrib_left],
+  rw [mem_factors (mul_ne_zero ha hb).bot_lt, mem_factors ha.bot_lt, mem_factors hb.bot_lt,
+    ←and_or_distrib_left],
   simpa only [and.congr_right_iff] using prime.dvd_mul
 end
 
 /-- If `a`,`b` are positive the prime divisors of `(a * b)` are the union of those of `a` and `b` -/
-lemma factors_mul_of_pos {a b : ℕ} (ha : 0 < a) (hb : 0 < b) :
+lemma factors_mul_to_finset_of_ne_zero {a b : ℕ} (ha : a ≠ 0) (hb : b ≠ 0) :
```

Comment:
i think in general there's really no need for spelling out `ne_zero` every time; there's no other real possible generality for this lemma

Reply:
Fair point, I'll remove